expr/datamap.go

```diff
@@ -25,3 +30,29 @@ func (dm DataMap) Clean() {
 		}
 	}
 }
+
+// CheckForOverlappingPoints runs through all series in the pool and makes sure there
+// are no series that are overlapping (otherwise returning them would cause issues)
+// This is not efficient and should probably only be called from tests
```

Comment:
overlap is ambiguous here. it is easy to lead people to think about situations like
```
  ------
      -------
```
What we're really checking for here, is whether any series have pointslices backed by the same array.

Fun fact: it is in theory fine to have 2 series backed by the same array but use non-overlapping sections of that array. as long as we don't grow them to where we start overlapping. But I agree with your code that for simplicity we should also reject such cases.

Reply:
actually, thinking about this again, the name overlap is apt.
because the kind of pattern shown is exactly what you can get into with same-array-backed slices.
So let's keep the name